stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -136,6 +136,11 @@ end
 function Base.replace_in_print_matrix(A::Diagonal,i::Integer,j::Integer,s::AbstractString)
     i==j ? s : Base.replace_with_centered_mark(s)
 end
+function Base.show(io::IO, A::Diagonal)
```

Comment:
Should we add similar methods to the other banded matrix types?

Reply:
Could do, but maybe not this PR?

For the more complicated ones, there are choices like whether to print `SymTridiagonal([1,2,3], [4,5])` or rather `SymTridiagonal([1 4 0; 4 2 5; 0 5 3])`.

Reply:
Xref  #55347 for Bidiagonal.